spaceship.zsh

```diff
@@ -43,6 +43,7 @@ if [ -z "$SPACESHIP_PROMPT_ORDER" ]; then
     time          # Time stampts section
     user          # Username section
     dir           # Current directory section
+    write_perm    # Write permission in current directory
```

Comment:
It doesn't have to be a separate section, Can we combine this `dir` section ?

Reply:
I like this idea!

Reply:
Another reason why I think it would be nice to have this as part of the `dir` section is because I'd like to move the "lock sign" closer to the directory name:

![image](https://user-images.githubusercontent.com/1177900/35197108-5df703c0-fedb-11e7-86b8-217936ac1e2a.png)

Reply:
Like this ?
```zsh
spaceship_dir() {
  [[ $SPACESHIP_DIR_SHOW == false ]] && return

  local dir

  # Threat repo root as a top-level directory or not
  if [[ $SPACESHIP_DIR_TRUNC_REPO == true ]] && spaceship::is_git; then
    local git_root=$(git rev-parse --show-toplevel)
    dir="$git_root:t${$(expr $(pwd) : "$git_root\(.*\)")}"
  else
    dir="%${SPACESHIP_DIR_TRUNC}~"
  fi

  spaceship::section \
    "$SPACESHIP_DIR_COLOR" \
    "$SPACESHIP_DIR_PREFIX" \
    "$dir" \
    "$SPACESHIP_DIR_SUFFIX"

  [[ $SPACESHIP_WRITE_PERM_SHOW == false ]] && return

  [[ -w . ]] && return

  spaceship::section \
    "$SPACESHIP_WRITE_PERM_COLOR" \
    "$SPACESHIP_WRITE_PERM_PREFIX" \
    "$SPACESHIP_WRITE_PERM_SYMBOL" \
    "$SPACESHIP_WRITE_PERM_SUFFIX"
}

```

Reply:
Such approach won't allow moving `PERM_SYMBOL` closer to the directory name, but I can live without it, I think I'll be using Powerline's lock symbol and it looks better with a space anyway.

![image](https://user-images.githubusercontent.com/1177900/35197651-878e5614-fee2-11e7-9e23-364086b11061.png)

Wait for @salmanulfarzy to reply if this is what he had in mind code-wise though.

Reply:
Do we need all this options ? In my opinion, Only `SPACESHIP_WRITE_PERM_COLOR` and `SPACESHIP_WRITE_PERM_SYMBOL` are needed. I would rather replace/append `lock` symbols to `SPACESHIP_DIR_SUFFIX` than making it a separate section. Altering `SPACESHIP_DIR_PREFIX` might not be visible if `dir` is first section, Which is common. 

```sh

SPACESHIP_DIR_WRITE_LOCK_SYMBOL="${SPACESHIP_DIR_WRITE_LOCK_SYMBOL=" 🔒 "}"

# Before invoking the section
if [[ ! -w . ]]; then
  SPACESHIP_DIR_SUFFIX="${SPACESHIP_DIR_WRITE_LOCK_SYMBOL}"
fi
```

It doesn't allow to change colour of symbol, But I don't see myself changing colour of lock symbol. If needed we could make use of escape codes in assignment.

Reply:
I think it's better if we can change color of lock, to make it more visible...

Reply:
Scratch what I said. On further usage it's better as separate section for customisability.

Reply:
Now we have two options
-  Make it separate section. But still have this issue https://github.com/denysdovhan/spaceship-prompt/pull/340#discussion_r162826201
- If color is the only option missing, We could hack around it using, 

```sh
  if [[ ! -w . ]]; then
    SPACESHIP_DIR_SUFFIX="%F{$SPACESHIP_DIR_LOCK_COLOR}$SPACESHIP_DIR_WRITE_LOCK_SYMBOL%f"
  fi
```

![dir-write-lock](https://i.imgur.com/P8Pa6Rr.png)

I'm open to both options. Suggestion ?

---
Is it just me or the lock symbol stays opaque ? Tried with iTerm and Terminal.